clang/lib/Sema/SemaHLSL.cpp

```diff
@@ -1033,12 +1033,15 @@ void SemaHLSL::handleRootSignatureAttr(Decl *D, const ParsedAttr &AL) {
 void SemaHLSL::handleNumThreadsAttr(Decl *D, const ParsedAttr &AL) {
   llvm::VersionTuple SMVersion =
       getASTContext().getTargetInfo().getTriple().getOSVersion();
+  bool IsDXIL = getASTContext().getTargetInfo().getTriple().getArch() ==
+                llvm::Triple::dxil;
+
   uint32_t ZMax = 1024;
```

Comment:
Shouldn't we even disable any Xmax/Ymax/Zmax check when targeting SPIR-V/Vk? Seems like those are set by the device limits, so not limited by the vk version

Reply:
I don't think so. These are close to the limits on actual machines. For Android, the existing checks are way beyond the limit.

See maxComputeWorkGroupInvocations and  maxComputeWorkGroupSize in https://vulkan.lunarg.com/doc/sdk/1.4.313.2/windows/profiles_definitions.html.